### PR TITLE
INFRANG-5290 - CI mini app w/ docker app support

### PR DESCRIPTION
**Pull request by andruwm**
This PR adds MVP support for detecting applications within a git repo in CI, building any Dockerfiles, and then pushing those tagged images to a private ECR repository. There are a number of TODO's regarding additional features which will be addressed in subsequent PRs.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	golang.org/x/net v0.7.0 // indirect
+	golang.org/x/sys v0.6.0 // indirect
+	golang.org/x/time v0.3.0 // indirect
+	golang.org/x/tools v0.5.0 // indirect
+	gopkg.in/yaml.v2 v2.4.0 // indirect
+	gopkg.in/yaml.v3 v3.0.1 // indirect
+	gotest.tools/v3 v3.4.0 // indirect
+)
+
+replace github.com/Clever/catapult/gen-go/models => /Users/andrew.marine/go/src/github.com/Clever/catapult/gen-go/models
```
Comment (andruwm) [inline go.mod:67]: Leave these catapult model changes local until I finalize this binary at which point I will make a catapult PR.

File: cmd/goci/main.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+func awsCfg(ctx context.Context) (aws.Config, error) {
+	opts := []func(*config.LoadOptions) error{
+		config.WithRegion("us-west-1"),
+	}
+
+	// In local environment we use the default credentials chain that
+	// will automatically pull creds from saml2aws,
+	if !environment.Local {
```
Comment (tnsardesai) [inline cmd/goci/main.go:81]: LGTM but would love a readme on how to run this locally to 
- test app
- to run in case CI is not working

**Review by tnsardesai — APPROVED** [review]

File: cmd/goci/main.go (hunk 2 above)
Comment (andruwm) [reply cmd/goci/main.go:81]: Good callout, would it be ok to include this in the followup PR?

**Review by tnsardesai — APPROVED** [review]
LGTM. maybe consider adding this to the application object and the hardcoded models too

**Pull request merged**